### ISSUE TITLE
[java] False negative on DontCallThreadRun when extending Thread

### ISSUE THREAD

**Issue by GiantCrocodile** [pmd/pmd#565]
There should be a rule to warn about using thread's run method and not using the start method. This can cause issues for coder who don't have much experience with Threads yet.

**Comment by jsotuyod** [comment]
@GiantCrocodile how would this be any different from [`java-basic/DontCallThreadRun`](https://pmd.github.io/pmd-5.8.1/pmd-java/rules/java/basic.html#DontCallThreadRun)?

Do you have any issues / false negatives with that rule?

**Comment by GiantCrocodile** [comment]
Thanks for your reply @jsotuyod. I ran the ruleset `all-java` and this code caused no error/warning with latest version:

```java
public class Test {
	public static void main(String[] args) {
		new TestThread().run();
	}
	
}
```

```java
public class TestThread extends Thread {
	@Override
	public void run() {
		System.out.println("test");
	}
}
```

**Comment by jsotuyod** [comment]
@GiantCrocodile ok, so it's a false negative then. Sounds like this rule was never updated to deal with type resolution given your example.

**Comment by GiantCrocodile** [comment]
I think this can be extended: You should usually implement Runnable, not extending Thread. You don't change/extend thread's behavior but you implement it.

**Comment by jsotuyod** [comment]
@GiantCrocodile I'm fixing the false positive on 6.0.0, and added a new issue to track the new rule request (a nice one by the way!), as #595
